ADR/0032-decoupling-deployment.md

```diff
@@ -0,0 +1,211 @@
+deployment environments of their choice, manually laying out their application resources in the repo
+(assisted by tools like `kam`), specifying image references by tag to match the `:released` or
+`:validated` tagging schem mentioned above, configuring Argo to deploy from their gitops repo, and
+configuring [renovatebot] to propagate image updates by digest to their gitops repo.
+
+**For manual creation of new environments** - the user manages this directly using a combination of
+their gitops repo and argo, outside of the AppStudio member cluster.
+
+**For automated testing in ephemeral environments** - the user specifies an
+[IntegrationTestScenario] CR with a [DeploymentTargetClass] reference. After a build completes, but
+before it executes tests, the integration-service creates a new [DeploymentTargetClaim] CR with
+a [DTCLS] matching the one on the [IntegrationTestScenario]. This should cause the
+[deployment-target-operator] to provision and bind a [DeploymentTarget] matching the
+[DeploymentTargetClass]. [integration-service] then passes the kubeconfig and [Snapshot] to the
+integration test to deploy the user's app and run tests.  The integration-service should delete the
+[DTC] once it isn’t needed anymore for the test.
+
+```mermaid
+flowchart TD
+    User --> |provides| IntegrationTestScenario
+    IntegrationTestScenario --> |references| DeploymentTargetClass
+    commit[fa:fa-code-commit Git Commit] --> |webhook| PipelinesAsCode
+    PipelinesAsCode --> |creates| build[Build PipelineRun]
+    build --> |triggers| integration-service
+    integration-service --> |consults| IntegrationTestScenario
+    integration-service --> |creates| DeploymentTargetClaim
+    DeploymentTargetClaim --> |references| DeploymentTargetClass
+    DeploymentTargetClaim --> |is bound to| DeploymentTarget
+    DeploymentTarget --> |is associated with| kubeconfig[kubeconfig Secret]
+    kubeconfig --> |is provided to| test
+    integration-service --> |creates| test[Test PipelineRun]
+    DeploymentTarget --> |represents| cluster
+    test --> |deploys user app to| cluster
+    test --> |executes tests against| cluster
+```
+
+## Consequences
+
+- Users who expect effortless deployment of their app when onboarding to the system will be
+  dissapointed. They have more work to do to set up a deployment of their app outside the system.
+- Users will lose visibility of their applications' deployments and status in the AppStudio UI
+  (HAC). Other systems like the Argo UI are arguably better at this than we are.
+- Users who expect to provide and manage their own resources to control their app will be delighted.
+  They now no longer have to interact with an intermediary API to try to express details about their
+  deployment(s).
+- As a team, we'll be in a position to try to achieve independence for the [DeploymentTarget] APIs,
+  make it usable outside the context of AppStudio, and ideally make it attractive for collaborators.
+
+## Phases of implementation
```

Comment:
Some of those phases can be done at the same time.

Reply:
Mentioned in 382bcb9.